### PR TITLE
Custom color palette: add default color name

### LINKED ISSUE #36473: Add nudge to include name for Custom Color Palette
## What problem does this address?

This is pulled from the [eleventh call for testing for the FSE Outreach Program](https://make.wordpress.org/test/2021/11/08/fse-program-testing-call-11-site-editing-safari/#comment-2217):

>I actually missed that I needed to add a name to the color at all 😅 Since there’s no placeholder text in the area where the color name should be added, I completely overlooked it and assumed I would just add the color, click Done, and voilà! However, it looks like not adding a color name at all means the colors won’t get saved. Adding some directive placeholder text next to the colors – or even an error message after clicking Done – might have helped me move past that. 

& another person:

> Just like @evarlese observed (and thanks for the precious notes!), the way to deal with the Custom Color Palette is not straight forward, at all. She already explained it really well, so adding my +1 also to the proposed solutions. Kudos Erica!

https://user-images.githubusercontent.com/26996883/141694781-83192cfb-0e89-4e4f-8268-6bb19af6f5fe.mp4

## What is your proposed solution?

Rather than resetting the custom color and not saving it, add in a prompt or warning to have someone include a name for the color. Perhaps it could be as simple as highlighting the field if it's empty or adding in placeholder text for where the name needs to go:

Screen Shot 2021-11-14 at 2 07 22 PM

To make it less jarring, could just add a weighted border that's black to point it out more as a field: 

Screen Shot 2021-11-14 at 2 08 17 PM
 
I'll leave it to designers :D

### PR DESCRIPTION
**Pull request by ramonjd**
(Maybe) Resolves https://github.com/WordPress/gutenberg/issues/36473

## Description

Creating a temporary name when we add a new color to the custom palette.

Screen Shot 2021-11-29 at 3 08 36 pm

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/src/palette-edit/index.js [hunk 1]
```diff
@@ -270,13 +271,18 @@ export default function PaletteEdit( {
 									: __( 'Add color' )
 							}
 							onClick={ () => {
+								const tempOptionName = sprintf(
+									/* translators: %s: is a temporary id for a custom color */
```
Comment (mtias) [inline packages/components/src/palette-edit/index.js:275]:
```suggestion
									/* translators: %s: is a temporary label for a custom color */
```

**Review by mtias — APPROVED** [review]
Looks good to me, nice quality of life change

**Comment by javierarce** [comment]
Thanks, @ramonjd, this works well.

I've noticed that the text strings include whitespace at the end (e.g.: "Color 5 "). Is that on purpose? If there's not a reason for that, could you remove it?

File: packages/components/src/palette-edit/index.js [hunk 2]
```diff
@@ -270,13 +271,18 @@ export default function PaletteEdit( {
...
+									__( 'Color %s ' ),
```
Comment (mtias) [inline packages/components/src/palette-edit/index.js:276]:
```suggestion
									__( 'Color %s' ),
```

**Comment by ramonjd** [comment]
> I've noticed that the text strings include whitespace at the end (e.g.: "Color 5 "). Is that on purpose? If there's not a reason for that, could you remove it?

No reason other than this PR is still WIP and was slapped together without much testing  😆 

Thanks for testing @javierarce and for the commit @mtias 🙇

**Comment by ramonjd** [comment]
Looking at the Gutenberg commit history, the [failing test](https://github.com/WordPress/gutenberg/runs/4358826442?check_suite_focus=true) for `packages/e2e-tests/specs/editor/blocks/image.test.js` is failing frequently. Looks like a network error. Pretty sure it's unrelated to this PR.

**Comment by noisysocks** [comment]
I'd like to get this into 5.9 beta 2. It's marked as an enhancement but it fixes a common source of frustration and isn't a very large code change.

**Comment by ramonjd** [comment]
> I'd like to get this into 5.9 beta 2. It's marked as an enhancement but it fixes a common source of frustration and isn't a very large code change.

👍 

We can do the smaller UI change in a separate PR: 

> Another important thing I would add right away is the hover state for the input fields when they aren't active:

From https://github.com/WordPress/gutenberg/issues/36473#issuecomment-971493346

**Pull request merged**